src/main/java/spoon/reflect/visitor/CommentHelper.java

```diff
@@ -83,7 +78,8 @@ static void printComment(PrinterHelper printer, CtComment comment) {
 	static void printCommentContent(PrinterHelper printer, CtComment comment, Function<String, String> transfo) {
 		CtComment.CommentType commentType = comment.getCommentType();
 		String content = comment.getContent();
-		String[] lines = LINE_SEPARATORS_RE.split(content);
+
+		String[] lines = content.lines().toArray(String[]::new);
```

Comment:
How about making this a forEach?

```suggestion
		content.lines().forEach(line -> {
			if (commentType == CtComment.CommentType.BLOCK) {
				printer.write(line);
				if (lines.length > 1) {
					printer.write(CtComment.LINE_SEPARATOR);
				}
			} else {
				printer.write(transfo.apply(line)).writeln(); // removing spaces at the end of the space
			}
		});
```